src/commands/timer.ts

```diff
@@ -0,0 +1,58 @@
+import BaseCommand, { CommandArgs } from "./base_command";
+import { getDebugContext, sendOptionsMessage } from "../helpers/discord_utils";
+import { GameOption, getGuildPreference } from "../helpers/game_utils";
+import _logger from "../logger";
+const logger = _logger("limit");
+
+export default class TimerCommand implements BaseCommand {
+    async call({ message, parsedMessage, gameSessions }: CommandArgs) {
+        const guildPreference = await getGuildPreference(message.guildID);
+        const gameSession = gameSessions[message.guildID];
+        if (parsedMessage.components.length === 0) {
+            guildPreference.resetTimer();
+            if (gameSession) {
+                gameSession.stopTimer();
+                gameSession.timerVal = null;
+            }
+            await sendOptionsMessage(message, guildPreference, GameOption.TIMER);
+            logger.info(`${getDebugContext(message)} | Timer disabled.`);
+            return;
+        }
+        const time = parseInt(parsedMessage.components[0]);
+
+        guildPreference.setTimer(time);
+        if (gameSession) {
+            gameSession.timerVal = time;
```

Comment:
Why does this need to be set on `gameSession` instead of read from `guildPreference`

Reply:
I rewrote some stuff and don't need it anymore, thanks for pointing out that I could read from `guildPreference`